### PR TITLE
Add swipe capabilities to image generation (regenerates same prompt with random seed)

### PR DESCRIPTION
**Pull request by edk208**
## Checklist:

- [ x ] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

_What is the reason for a change?_
Re-rolling image generation by the swipe process is a convenient way to regenerate an image via the image prompt, with a different seed.  This is also a requested feature [#1825] maybe also [#2576]

_What did you do to achieve this?_
Created a new function called reGeneratePicture that takes the existing prompt and sends it to an image generation backend with a random seed.  When re-rolling the prompt _is not regenerated_ in this case; it is using the same prompt with a different (random) seed.  I think this makes the most sense as re-rolling the image generation with prompt generation could lead to a very different response.  

For the case that you want a whole new image prompt, you would probably just generate as you usually would (and not utilize the swipe capability). 

_How would a reviewer test the change?_
Choose your favorite image generation extension and generate an image as usual.  The swipes will pop up and you can swipe right and left as usual.

### CODE REVIEW

File: public/script.js [hunk 1]
```diff
@@ -133,6 +133,10 @@ import {
     MIN_LENGTH,
 } from './scripts/horde.js';
 
+import {
```
Comment (Cohee1207) [inline public/script.js]: Importing from an extension script would make it impossible to disable the extension. Consider using eventSource events instead.

Comment (edk208) [reply public/script.js]: ok i'll look into it and refactor

**Comment by Wolfsblvt** [comment]
Just a request/idea, not needed for this PR. But if you add this ability as button/swipes, it would be cool to expand the `/sd` slash command too, likely with an optional parameter, or add a new one.
So one could use STscript to generate multiple picutes at once, etc.

File: public/script.js (hunk 1 above)
Comment (edk208) [reply public/script.js]: updated for eventSource

**Comment by edk208** [comment]
> Just a request/idea, not needed for this PR. But if you add this ability as button/swipes, it would be cool to expand the `/sd` slash command too, likely with an optional parameter, or add a new one. So one could use STscript to generate multiple picutes at once, etc.

thanks for the suggestion.  I spent some time on the slash commands and, while I got something working, it is a bit hacky. 
 I think I will spend more time on it in a separate PR

**Comment by Cohee1207** [comment]
There's a problem I'd like to address after trying it up: when you add an image to a message (either by paintbrush icon or attaching to a message), you'll no longer be able to swipe it normally. I imagine image swipes being decoupled from text swipes. For example, keeping an array of image URLs in the "extra" message metadata object and adding a couple of buttons to cycle through them. I can attempt that in a separate PR.

**Comment by Cohee1207** [comment]
Check this out: #2655

**Comment by edk208** [comment]
redundant with [#2655], closing.

**Pull request closed without merging**